### PR TITLE
feat: add renku refresh token for data service

### PR DESCRIPTION
**Pull request by olevski**
Also adds the anonymous user ID. All of this is required to be able to launch sessions with the new operator for the data service.

/deploy

### CODE REVIEW

File: internal/revproxy/auth.go [hunk 1]
```diff
@@ -357,3 +358,47 @@ var gitlabCliTokenInjector TokenInjector = func(c echo.Context, accessToken mode
...
+		expiresAt = accessToken.ExpiresAt.Unix()
+	}
+	slog.Debug(
+		"PROXY AUTH MIDDLEWARE",
+		"message",
+		"injected token in header",
+		"header",
+		headerKey,
+		"token",
+		accessToken.String(),
```
Comment (Panaetius) [inline internal/revproxy/auth.go:395]: same here

**Review by Panaetius — DISMISSED** [review]

File: internal/revproxy/auth.go [hunk 2]
```diff
@@ -357,3 +358,47 @@ var gitlabCliTokenInjector TokenInjector = func(c echo.Context, accessToken mode
...
+	existingToken := c.Request().Header.Get(headerKey)
+	if existingToken != "" {
+		slog.Debug(
+			"PROXY AUTH MIDDLEWARE",
+			"message",
+			"token already present in header, skipping",
+			"header",
+			headerKey,
+			"token",
+			accessToken.String(),
```
[the remark this answers is not in the document]
Comment (leafty) [reply internal/revproxy/auth.go:375]: See the `.String()` method:
https://github.com/SwissDataScienceCenter/renku-gateway/blob/77055670146807e02286e5c53fd5173192c6cd53/internal/models/token.go#L49-L60

The value of the token is not printed in the logs.

[the remark this answers is not in the document]
Comment (olevski) [reply internal/revproxy/auth.go:375]: As @leafty mentioned the `String()` method on the tokens is set to just print `<redacted>` for the sensitive parts. This is the same as setting `__repr__` on a python object.

File: internal/revproxy/auth.go (hunk 1 above)
Comment (olevski) [reply internal/revproxy/auth.go:395]: same as above the sensitive part will not show up.

**Comment by olevski** [comment]
This change adds additional credentials in the headers to the data service coming from the gateway. It does not break or remove anything that currently is done. As such it can be merged and published before the jupyter free sessions are released.

**Review by leafty — DISMISSED** [review]
LGTM, just one suggestion.

File: internal/revproxy/auth.go [hunk 3]
```diff
@@ -357,3 +358,47 @@ var gitlabCliTokenInjector TokenInjector = func(c echo.Context, accessToken mode
...
+			"requestID",
+			utils.GetRequestID(c),
+		)
+		return nil
+	}
+
+	// NOTE: As long as the token comes from the database we can trust it and do not have to validate it.
+	// Each service that the request ultimately goes to will also validate before it uses the token
+	var expiresAt int64 = -1
```
[the remark this answers is not in the document]
Comment (olevski) [reply internal/revproxy/auth.go]: That makes perfect sense. Not sure what I was thinking.

**Review by leafty — APPROVED** [review]

**Pull request merged**